app/controllers/api/automate_controller.rb

```diff
@@ -0,0 +1,34 @@
+module Api
+  class AutomateController < BaseController
+    def show
+      ae_browser = MiqAeBrowser.new(@auth_user_obj)
+      begin
+        resources = ae_browser.search(@req.c_suffix, ae_search_options)
+      rescue => err
+        raise BadRequestError, err.to_s
+      end
+      attributes = params['attributes'] ? %w(fqname) | params['attributes'].to_s.split(',') : nil
+      resources = resources.collect { |resource| filter_ae_resource(resource, attributes) } if attributes
+      render_resource :automate, :name => "automate", :subcount => resources.count, :resources => resources
+    end
+
+    private
+
+    def filter_ae_resource(resource, attributes = nil)
```

Comment:
good point, I had updated the caller to not call it if there were not attributes, (not need to do a noop collect on potentially thousands of objects).  will simplify. Thanks.